epoxy-adapter/src/main/java/com/airbnb/epoxy/stickyheader/StickyHeaderLinearLayoutManager.kt

```diff
@@ -0,0 +1,621 @@
+package com.airbnb.epoxy.stickyheader
+
+import android.content.Context
+import android.graphics.PointF
+import android.os.Build
+import android.os.Parcelable
+import android.view.View
+import android.view.ViewGroup
+import android.view.ViewTreeObserver
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import com.airbnb.epoxy.BaseEpoxyAdapter
+import kotlinx.android.parcel.Parcelize
+
+/**
+ * Adds sticky headers capabilities to your [RecyclerView.Adapter].
+ * The adapter must implement [HasStickyHeader] to indicate which items are
+ * sticky.
+ */
+class StickyHeaderLinearLayoutManager @JvmOverloads constructor(
+    context: Context,
+    orientation: Int = RecyclerView.VERTICAL,
+    reverseLayout: Boolean = false
+) : LinearLayoutManager(context, orientation, reverseLayout) {
+
+    private lateinit var adapter: BaseEpoxyAdapter
+
+    // Translation for header
+    private var translationX: Float = 0f
+    private var translationY: Float = 0f
+
+    // Header positions for the currently displayed list and their observer.
+    private val headerPositions = mutableListOf<Int>()
+    private val headerPositionsObserver = HeaderPositionsAdapterDataObserver()
+
+    // Sticky header's ViewHolder and dirty state.
+    private var stickyHeader: View? = null
+    private var stickyHeaderPosition = RecyclerView.NO_POSITION
+
+    // Save / Restore scroll state
+    private var scrollPosition = RecyclerView.NO_POSITION
+    private var scrollOffset = 0
+
+    override fun onAttachedToWindow(recyclerView: RecyclerView) {
+        super.onAttachedToWindow(recyclerView)
+        recyclerView.adapter?.let(this::setAdapter)
+    }
+
+    override fun onAdapterChanged(oldAdapter: RecyclerView.Adapter<*>?, newAdapter: RecyclerView.Adapter<*>?) {
+        super.onAdapterChanged(oldAdapter, newAdapter)
+        newAdapter?.let(this::setAdapter)
```

Comment:
should anything be cleared when this newAdapter is null?

Reply:
Good point. Now that I think about it, yes we should be clearing few things if the adapter is null.
I'll update it 👍